src/ott/solvers/linear/continuous_barycenter.py

```diff
@@ -107,13 +112,73 @@ def solve_linear_ot(
     )(bar_prob.weights, barycenters_per_measure)
 
     return self.set(
+        x=x_new,
         costs=updated_costs,
         linear_convergence=linear_convergence,
-        errors=errors,
-        x=x_new
+        linear_outputs=outs,
+        errors=errors
     )
 
 
+class FreeBarycenterOutput(NamedTuple):
+  """Holds the output of a Free Wasserstein barycenter solver.
+
+  Objects of this class contain both solutions and problem definition of a
+  regularized Free Barycenter OT problem, along several methods that can be used
+  to access its content.
+
+  Args:
+    x: barycenter points.
+    a: barycenter weights.
+    costs: Holds the sequence of regularized GW costs seen through the outer
+      loop of the solver.
+    linear_convergence: Holds the sequence of bool convergence flags of the
+      inner Sinkhorn iterations.
+    linear_outputs: Holds latest output objects returned by linear solver when
+      recomputing transport between barycenter vs. each of the input measures.
+    errors: Holds sequence of vectors of errors of the Sinkhorn algorithm
+      at each iteration.
+  """
+
+  x: jnp.ndarray = None
+  a: jnp.ndarray = None
+  bar_prob: barycenter_problem.FreeBarycenterProblem = None
+  costs: jnp.ndarray = None
+  linear_convergence: jnp.ndarray = None
+  linear_outputs: Optional[Tuple[linear_output_t, ...]] = None
+  errors: Optional[jnp.ndarray] = None
+
+  @property
+  def all_linear_solvers_converged(self) -> bool:
+    """Whether all linear convergence flags converged."""
+    return jnp.all(self.linear_convergence[self.linear_convergence != -1])
+
+  def matrix_at_index(self, measure_index: int) -> jnp.ndarray:
+    """Return the transport matrix from barycenter to measure_index measure."""
+    size_measure = self.bar_prob.num_per_measure[measure_index]
+    matrix = self.linear_output_at_index(measure_index).matrix
+    return jax.lax.dynamic_slice(matrix, (0, 0), (self.bar_size, size_measure))
```

Comment:
Afaik we don't need to slice on `axis=0`, would prefer to use `jax.lax.dynamic_slice_in_dim` to be more explicit.